### DISCUSSION TITLE
Build error occured

### DISCUSSION THREAD

**Discussion by vagyvin** [graphcommerce-org/graphcommerce#1389] [Q&A]
### Provide environment information

MacOS Monterey

### How are you deploying your application? (if relevant)

yarn build

### Describe the Bug

Getting a build error

/account/contact
/account/name
/500
info  - Generating static pages (45/45)

> Build error occurred
[Error: ENOENT: no such file or directory, rename '.next/export/en/product.html' -> '.next/server/pages/en/product.html'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'rename',
  path: '.next/export/en/product.html',
  dest: '.next/server/pages/en/product.html'
}

### Expected Behavior

Should build successfully

### To Reproduce

yarn build

**Comment by paales** [comment]
Hi @vagyvin, can you please try and delete the `.next` folder and try again?

**Comment by vagyvin** [comment]
Hi,

I tried that. Didn't work. I believe it has to do something with the locale.
I was trying to get rid of the locale. "/en" since we don't use that for our Magento store. It's default set to English
Do you know how I can get rid of the "/en" for url's without causing any issues?

Appreciate you guys helping out.

**Comment by paales** [comment]
Does the documentation for `NEXT_PUBLIC_LOCALE_STORES` explain it?
https://www.graphcommerce.org/docs/getting-started/create#configuration

**Comment by vagyvin** [comment]
Looks like the documentation is updated

I tried with the most recent git pull with .env setting as follows
NEXT_PUBLIC_LOCALE_STORES='{"en-us":"default"}'

I'm seeing the following error. I can see that the url includes "/en-us"
Error occurred prerendering page "/en-us/product". Read more: https://nextjs.org/docs/messages/prerender-error
Error: The provided export path '/product' doesn't match the '/product/[url]' page.

Error occurred prerendering page "/en-us/computers-tablets". Read more: https://nextjs.org/docs/messages/prerender-error
Error: filter created_at with FilterTypeInput FilterTypeInput not implemented

> Build error occurred
Error: Export encountered errors on following paths:
        /[...url]: /en-us/computers-tablets
        /product/[url]: /en-us/product
        /search/[[...url]]: /en-us/search

**Comment by paales** [comment]
When you use only _one_ locale you should not see any segments in the URL. In the default example we have the following:

```
NEXT_PUBLIC_LOCALE_STORES='{"en-us": "en_US", "nl": "nl_NL", "fr-be": "fr_BE", "nl-be": "nl_BE", "en-gb": "en_GB", "en-ca": "en_CA"}'
```

This results in no added URL key for the 'en_US' storeview: https://graphcommerce.vercel.app/women/business.

**Comment by vagyvin** [comment]
I'm getting this error 

Error: filter created_at with FilterTypeInput FilterTypeInput not implemented

**Comment by paales** [comment]
Could you try upgrading to the latest version, this has recently been fixed in this commit https://github.com/graphcommerce-org/graphcommerce/commit/9b3488c6a03cc09a647f43f6a8b36d96e97e5bb8

**Comment by vagyvin** [comment]
That issue is fixed but I still see the build error. 
yarn codegen and yarn dev works fine.

For locales and store_view I had to use 
NEXT_PUBLIC_LOCALE_STORES='{"en-us": "default"}' -----> Works
I tried NEXT_PUBLIC_LOCALE_STORES='{"en-us": "en_US"}' -----> doesn't work. 
NEXT_PUBLIC_LOCALE_STORES='{"en-us": "en_US", "nl": "nl_NL", "fr-be": "fr_BE", "nl-be": "nl_BE", "en-gb": "en_GB", "en-ca": "en_CA"}' -----> Doesn't work either

info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data  
[    ] info  - Generating static pages (0/45)
Error occurred prerendering page "/en-us/product". Read more: https://nextjs.org/docs/messages/prerender-error
Error: The provided export path '/product' doesn't match the '/product/[url]' page.

**Comment by paales** [comment]
```
NEXT_PUBLIC_LOCALE_STORES='{"en-us": "your_magento_store_code"}'
```

Replace your_magento_store_code with the store_code that your installation is using.

Search for 'If you need to fetch a list of available store_codes' on https://www.graphcommerce.org/docs/getting-started/create#configuration

**Comment by vagyvin** [comment]
I did 
This is what I get

{
  "data": {
    "availableStores": [
      {
        "store_code": "default",
        "store_name": "English"
      }
    ]
  },

NEXT_PUBLIC_LOCALE_STORES='{"en-us": "default"}'

**Comment by paales** [comment]
Yeahh! You then should use `NEXT_PUBLIC_LOCALE_STORES='{"en-us": "default"}'`! :)

**Comment by vagyvin** [comment]
Same error though

info - Creating an optimized production build
info - Compiled successfully
info - Collecting page data
[ ] info - Generating static pages (0/45)
Error occurred prerendering page "/en-us/product". Read more: https://nextjs.org/docs/messages/prerender-error
Error: The provided export path '/product' doesn't match the '/product/[url]' page.
